starcheck/pcad_att_check.py

```diff
@@ -35,8 +35,18 @@ def test_check_characteristics_date():
                                     '2015Nov02 at 00:00:00.000')
     assert ok is False
 
+def recent_history(time, file):
+    for state_line in reversed(open(file).readlines()):
```

Comment:
In this small routine `state_` everywhere is just adding noise because it isn't helping to understand what is in the line (yes, a "state", but what is that?).  It makes the code harder to read because the variable names are more difficult to visually distinguish (`state_match` vs. `state_line`).  

Let's shoot for a consistent pattern of using `line` for a line in a file or list of lines, and `match` for a regex match, `ok` for a one- or two-off boolean check, and putting a comment line describing things.  (And of course don't use this pattern when it really doesn't make sense, perhaps when the context is more than a screenfull of code lines).